packages/components/src/utils/hooks/test/use-focus-outside.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import TestUtils from 'react-dom/test-utils';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useFocusOutside } from '../';
+import ReactDOM from 'react-dom';
+
+let wrapper, onFocusOutside;
+
+describe( 'useFocusOutside', () => {
+	let origHasFocus;
+
+	const FocusOutsideComponent = ( { onFocusOutside: callback } ) => (
+		<div { ...useFocusOutside( callback ) }>
+			<input />
+			<input type="button" />
+		</div>
+	);
+
+	// this is needed because TestUtils does not accept a stateless component.
+	// anything run through a HOC ends up as a stateless component.
+	const getTestComponent = ( WrappedComponent, props ) => {
+		class TestComponent extends Component {
+			render() {
+				return <WrappedComponent { ...props } />;
+			}
+		}
+		return <TestComponent />;
+	};
+
+	const simulateEvent = ( event, index = 0 ) => {
+		const element = TestUtils.scryRenderedDOMComponentsWithTag(
+			wrapper,
+			'input'
+		);
+		TestUtils.Simulate[ event ]( element[ index ] );
+	};
+
+	beforeEach( () => {
+		// Mock document.hasFocus() to always be true for testing
+		// note: we overide this for some tests.
+		origHasFocus = document.hasFocus;
+		document.hasFocus = () => true;
```

Comment:
Nitpick: we could just use `jest.spyOn` to simplify some code.
```suggestion
		// Mock document.hasFocus() to always be true for testing
		// note: we overide this for some tests.
		jest.spyOn(document, 'hasFocus')
			.mockImplementation(() => true);
```

And later we can just...

```js
document.hasFocus.mockRestore();
```

We can also temporary mock it to another value in individual test

```js
document.hasFocus.mockImplementationOnce(() => false);
```

This method requires `document.hasFocus` to be exist though, not sure if `jsdom` has it or not.

Reply:
Ah, yeah, I don't want to mess with these tests too much right now.